assets/blocks/quiz/data.js

```diff
@@ -52,24 +53,51 @@ export function syncQuestionBlocks( structure, blocks ) {
 
 		let block = blocks ? findQuestionBlock( blocks, item ) : null;
 
-		const innerBlocks =
-			( description && rawHandler( { HTML: description } ) ) ||
-			block?.innerBlocks;
-
 		if ( ! block ) {
-			block = createBlock( questionBlock.name, attributes, innerBlocks );
+			block = createQuestionBlock( item );
 		} else {
 			block.attributes = {
 				...block.attributes,
 				...attributes,
 			};
-			block.innerBlocks = innerBlocks;
+
+			const innerBlocks =
+				( description && rawHandler( { HTML: description } ) ) || [];
+
```

Comment:
Bit of an edge case, but could be safer to run the incoming attributes through `prepareQuestionBlock` even when updating an existing block. Also moving the `rawHandler` parsing lines inside that function can eliminate some duplication here.

Reply:
I had it in there initially (removed in https://github.com/Automattic/sensei/pull/4064/commits/9882ccb42cb3699dd9910d373c33e94b15d8d999), but removed it because this code would get hit when:
1) In `post_content` on load, in which the deprecated parser in the block editor would take care of it.
2) On post save, when these transformations would have already happened.

Which edge case were you thinking as a possibility?

Reply:
The case could be when the lesson already has the quiz block, but question media was re-added by some plugin/filter/user switching to legacy editor. Probably fine to ignore.